lib/hal/led.h

```diff
@@ -0,0 +1,12 @@
+#ifndef HAL_LED_H
+#define HAL_LED_H
+
+#define LED_OFF    0x00
+#define LED_GREEN  0x01
+#define LED_RED    0x10
+#define LED_ORANGE 0x11
+
+void XSetAutoLED(void);
+void XSetManualLED(unsigned int t1, unsigned int t2, unsigned int t3, unsigned int t4);
```

Comment:
Please add a comment indicating what these parameters do: that t1-t4 correspond to the four LED states which are cycled repeatedly.